src/viewport_renderer.rs

```diff
@@ -665,3 +682,99 @@ fn wgpu_size_of<T>() -> wgpu::BufferAddress {
     wgpu::BufferAddress::try_from(size)
         .unwrap_or_else(|_| panic!("Size {} does not fit into wgpu BufferAddress", size))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn triangle() -> (Vec<[f32; 3]>, Vec<[f32; 3]>) {
+        #[rustfmt::skip]
+        let vertex_positions = vec![
+            [ -0.3, -0.5,  0.0 ],
+            [  0.3, -0.5,  0.0 ],
+            [  0.0,  0.5,  0.0 ],
+        ];
+
+        #[rustfmt::skip]
+        let vertex_normals = vec![
+            [ 0.0, 0.0, 1.0 ],
+            [ 0.0, 0.0, 1.0 ],
+            [ 0.0, 0.0, 1.0 ],
+        ];
+
+        (vertex_positions, vertex_normals)
+    }
+
+    fn triangle_indexed() -> (Vec<Index>, Vec<[f32; 3]>, Vec<[f32; 3]>) {
+        let (vertex_positions, vertex_normals) = triangle();
+        let indices = vec![0, 1, 2];
+
+        (indices, vertex_positions, vertex_normals)
+    }
+
+    #[test]
+    fn test_create_valid_geometry_does_not_panic() {
+        let (positions, normals) = triangle();
+        let _geometry = Geometry::from_positions_and_normals(positions, normals);
+
+        // Didn't panic!
+    }
+
+    #[test]
+    fn test_create_valid_indexed_geometry_does_not_panic() {
+        let (indices, positions, normals) = triangle_indexed();
+        let _geometry = Geometry::from_positions_and_normals_indexed(indices, positions, normals);
+
+        // Didn't panic!
```

Comment:
Just assert indices and vertex data in both of these tests. There's logic transforming incoming data into other data, so that's what should be tested here anyway. If it doesn't fail, it didn't panic, so that's tested as well.

I know you find it to be a noise, but you'll write it once and most likely never read it ever again unless there's a bug. If it was up to me, I'd create 4 tests with 1 assert each:

* `test_geometry_from_positions_and_normals_produces_valid_vertex_data`
* `test_geometry_from_positions_and_normals_produces_valid_indices`
* `test_geometry_from_positions_and_normals_indexed_produces_valid_vertex_data`
* `test_geometry_from_positions_and_normals_indexed_produces_valid_indices`

But I'm ok with 2 tests with 2 asserts each (asserting both vertex data and indices) if you don't want that much noise.

We most likely already spent more time discussing tests than it'd take to actually write or read them.

Reply:
Yeah, seems like the `assert!(1 == 1)` scenario to me, but I'll add those asserts. I'd keep the rest of the tests. Geometry has to be validated at some point, but the renderer seems like too late a place.

Reply:
Why `1 == 1`? It's input -> transformation -> output and you want to test if the transformation part does what it's supposed to do. It's not like the output is the same as input.